File: ios/src/com/bombbird/terminalcontrol/IOSLauncher.java
package com.bombbird.terminalcontrol;

import com.bombbird.terminalcontrol.utilities.ToastManager;
import org.robovm.apple.foundation.NSAutoreleasePool;
import org.robovm.apple.uikit.UIApplication;

import com.badlogic.gdx.backends.iosrobovm.IOSApplication;
import com.badlogic.gdx.backends.iosrobovm.IOSApplicationConfiguration;

public class IOSLauncher extends IOSApplication.Delegate {
    @Override
    protected IOSApplication createApplication() {
        IOSApplicationConfiguration config = new IOSApplicationConfiguration();
        return new IOSApplication(new TerminalControl(new TextToSpeechManager(), new ToastManager() {
            @Override
            public void saveFail() {
                //No default implementation
            }

            @Override
            public void readStorageFail() {
                //No default implementation
            }

            @Override
            public void jsonParseFail() {
                //No default implementation
            }
        }), config);
    }

    public static void main(String[] argv) {
        NSAutoreleasePool pool = new NSAutoreleasePool();
        UIApplication.main(argv, null, IOSLauncher.class);
        pool.close();
    }
}

File: ios/src/com/bombbird/terminalcontrol/IOSLauncher.kt
package com.bombbird.terminalcontrol

import com.badlogic.gdx.backends.iosrobovm.IOSApplication
import com.badlogic.gdx.backends.iosrobovm.IOSApplicationConfiguration
import com.bombbird.terminalcontrol.utilities.BrowserInterface
import com.bombbird.terminalcontrol.utilities.DiscordManager
import com.bombbird.terminalcontrol.utilities.PlayServicesInterface
import com.bombbird.terminalcontrol.utilities.ToastManager
import com.bombbird.terminalcontrol.utilities.files.ExternalFileHandler
import org.robovm.apple.foundation.NSAutoreleasePool
import org.robovm.apple.uikit.UIApplication

class IOSLauncher : IOSApplication.Delegate() {
    override fun createApplication(): IOSApplication {
        val config = IOSApplicationConfiguration()
        return IOSApplication(TerminalControl(
            TextToSpeechManager(), object : ToastManager {}, object : DiscordManager {}, object : ExternalFileHandler {}, object : BrowserInterface {}, object : PlayServicesInterface {}), config)
    }

    companion object {
        @JvmStatic
        fun main(argv: Array<String>) {
            val pool = NSAutoreleasePool()
            UIApplication.main<UIApplication, IOSLauncher>(argv, null, IOSLauncher::class.java)
            pool.close()
        }
    }
}